R/fabricate.R

```diff
@@ -152,24 +152,18 @@ fabricate_data_single_level <- function(data = NULL,
 
   if (!is.null(N)) {
     if (length(N) != 1) {
-      if(is.null(ID_label)) {
-        stop(
-          "At the top level, you must provide a single number to N."
-          )
-      } else {
-        stop(
-          "At the top level, ",
-          ID_label,
-          ", you must provide a single number to N."
-        )
-      }
-    }
-
-    if(is.numeric(N) & any(!N%%1 == 0)) {
-      stop(paste0(
+      stop(
+        "At the top level, ",
+        ifelse(!is.null(ID_label),
+               paste0(ID_label, ", "),
+               ""),
```

Comment:
No need to check null in this case, eg

```
> foo <- NULL
> stop("Foo is ", foo)
Error: Foo is 
```

Reply:
This was mostly about making clean, legible text. As-is the error was "At the top level, , you must provide..." and I thought the , , section was ugly. By ifelsing here, we avoid the grammar problem.